nucypher/datastore/dkg.py

```diff
@@ -13,31 +15,41 @@ def __init__(self):
     def store_transcript(self, ritual_id: int, transcript: Transcript) -> None:
         self.data["transcripts"][ritual_id] = bytes(transcript)
 
-    def get_transcript(self, ritual_id: int) -> Transcript:
+    def get_transcript(self, ritual_id: int) -> Optional[Transcript]:
         data = self.data["transcripts"][ritual_id]
         transcript = Transcript.from_bytes(data)
         return transcript
 
-    def store_transcript_receipt(self, ritual_id: int, receipt: TxReceipt) -> None:
-        self.data["transcript_receipts"][ritual_id] = receipt
+    def store_transcript_receipt(
+        self, ritual_id: int, txhash_or_receipt: Union[TxReceipt, HexBytes]
+    ) -> None:
+        self.data["transcript_receipts"][ritual_id] = txhash_or_receipt
 
-    def get_transcript_receipt(self, ritual_id: int) -> TxReceipt:
-        return self.data["transcript_receipts"][ritual_id]
+    def get_transcript_receipt(
+        self, ritual_id: int
+    ) -> Optional[Union[TxReceipt, HexBytes]]:
+        return self.data["transcript_receipts"].get(ritual_id)
```

Comment:
We use `.get()` because if the entry for that `ritual_id` is not present, we want it to return `None` and not raise a `KeyError`.

Note the typehint for what the method returns:
`Optional[Union[TxReceipt, HexBytes]]` which effectively means the method can return either `TxReceipt` OR `HexBytes` OR `None`.

Reply:
> Should we use .get() for the rest of calls to ["transcript_receipts"][ritual_id] to add homogeneity to the code? (like in lines 19, 34, 44, and so on)

You are absolutely right - I'll take care of that. Nice catch!

Reply:
✔️